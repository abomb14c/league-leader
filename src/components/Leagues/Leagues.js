import React from 'react';
import CreateLeague from '../../containers/CreateLeague/CreateLeague';
import './leagues.css';

const Leagues = () => {
  return (
    <div className="leagues-container">
      <CreateLeague />
    </div>
  );
};

export default Leagues; 

