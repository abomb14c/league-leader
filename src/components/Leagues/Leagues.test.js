import React from 'react';
import { shallow } from 'enzyme';
import  {Leagues, mapStateToProps}  from './Leagues';

describe('Leagues', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      user: {}
    };
    wrapper = shallow(<Leagues {...mockProps}/>, { disableLifecycleMethods: true });
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should map the user and EPL to props', () => {
    const mockState = {
      user: {user_id:1},
      EPL: [{}, {}, {}],
      draftTeams: [{}, {}, {}]
    };

    const expected = {
      user: {user_id:1}
    };
    
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});