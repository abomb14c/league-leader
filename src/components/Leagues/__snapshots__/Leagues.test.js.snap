// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Leagues
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <ExistingLeagues
          leagues={Array []}
/>,
      ],
      "className": "leagues-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "leagues": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <ExistingLeagues
            leagues={Array []}
/>,
        ],
        "className": "leagues-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "leagues": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
